core/src/test/java/io/grpc/ContextTest.java

```diff
@@ -726,4 +727,49 @@ public void execute(Runnable r) {
       runnables.add(r);
     }
   }
+
+  @Test
+  public void childContextListenerNotifiedAfterParentListener() {
+    final AtomicReference<LogRecord> logEntry = new AtomicReference<LogRecord>(null);
```

Comment:
Instead of trying to capture the logs, why not just move the assertFalse out of the listener?
```java
      final AtomicBoolean clientNotCalled = new AtomicBoolean();
      parent.addListener(new Context.CancellationListener() {
        @Override
        public void cancelled(Context context) {
          clientNotCalled.set(!childCalled.get());
        }
      }, MoreExecutors.directExecutor());
      ...
      assertTrue(clientNotCalled.get());
```

That seems like a _lot_ less boilerplate and less likelihood for a testing bug.